src/web/app/components/dialogs/CodeGenerator/Barcode.tsx

```diff
@@ -0,0 +1,130 @@
+import classNames from 'classnames';
+import JsBarcode, { Options } from 'jsbarcode';
+import React, { useEffect, useRef, useState, useCallback } from 'react';
+import S from './Barcode.module.scss';
+
+export type Renderer = 'canvas' | 'image' | 'svg';
+
+export type Format =
+  | 'CODE39'
+  | 'CODE128'
+  | 'CODE128A'
+  | 'CODE128B'
+  | 'CODE128C'
+  | 'EAN13'
+  | 'EAN8'
+  | 'EAN5'
+  | 'EAN2'
+  | 'UPC'
+  | 'UPCE'
+  | 'ITF14'
+  | 'ITF'
+  | 'MSI'
+  | 'MSI10'
+  | 'MSI11'
+  | 'MSI1010'
+  | 'MSI1110'
+  | 'pharmacode'
+  | 'codabar';
+
+export const formats: Format[] = [
+  'CODE39',
+  'CODE128',
+  'CODE128A',
+  'CODE128B',
+  'CODE128C',
+  'EAN13',
+  'EAN8',
+  'EAN5',
+  'EAN2',
+  'UPC',
+  'UPCE',
+  'ITF14',
+  'ITF',
+  'MSI',
+  'MSI10',
+  'MSI11',
+  'MSI1010',
+  'MSI1110',
+  'pharmacode',
+  'codabar',
+];
+
+export const defaultOptions: Options = {
+  format: 'CODE128',
+  width: 2,
+  height: 100,
+  displayValue: true,
+  fontOptions: '',
+  font: 'Noto Sans',
+  textAlign: 'center',
+  textPosition: 'bottom',
+  textMargin: 2,
+  fontSize: 20,
+  background: '#ffffff',
+  lineColor: '#000000',
+  margin: 10,
+  ean128: false,
+};
+
+export interface BarcodeProps {
+  renderer?: Renderer;
+  value: string;
+  options?: Options;
+  style?: React.CSSProperties;
+  className?: string;
+}
+
+export function Barcode({
+  style,
+  className,
+  value,
+  options = defaultOptions,
+  renderer = 'svg',
+}: Readonly<BarcodeProps>): JSX.Element {
+  const containerRef = useRef(null);
+  const [error, setError] = useState<string | null>(null);
+
+  const renderBarcode = useCallback(() => {
+    if (containerRef.current) {
+      try {
+        JsBarcode(containerRef.current, value, options);
+        setError(null);
+      } catch {
+        setError('The value is invalid for the selected format.');
+      }
+    }
+  }, [value, options]);
+
+  useEffect(() => {
+    renderBarcode();
+  }, [renderBarcode]);
```

Comment:
Maybe handle `renderBarcode` directly in `useEffect` instead of using two hooks?